airflow/models.py

```diff
@@ -3626,6 +3629,52 @@ def delete(cls, xcoms, session=None):
         session.commit()
 
 
+class DagStat(Base):
+    __tablename__ = "dag_stats"
+
+    dag_id = Column(String(ID_LEN), primary_key=True)
+    state = Column(String(50), primary_key=True)
+    count = Column(Integer, default=0)
+    dirty = Column(Boolean, default=False)
+
+    def __init__(self, dag_id, state, count, dirty=False):
+        self.dag_id = dag_id
+        self.state = state
+        self.count = count
+        self.dirty = dirty
+
+    @staticmethod
+    @provide_session
+    def set_dirty(dag_id, session=None):
+        for dag in session.query(DagStat).filter(DagStat.dag_id == dag_id):
+            dag.dirty = True
+        session.commit()
+
+    @staticmethod
+    @provide_session
+    def clean_dirty(session=None):
```

Comment:
Since we now process each DAG in a separate per-file scheduler job, we cannot assume a single schedule job anymore and this code will no longer be thread-safe or efficient. Instead, where the jobs.py currently calls manage_slas(), you can execute a clean_dirty method for a set of dags.